### PR TITLE
[TRANSFORMATIONS][GPU] Fix bool type of attention mask decomposition issue in ScaledDotProductAttentionDecomposition

### PR DESCRIPTION
**Pull request by andrew-k-park**
### Details:
 - In `ScaledDotProductAttentionDecomposition`, the accuracy issue was fixed because the attention mask decomposition with bool type was not aligned with [pytorch impl](https://docs.pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html)
 - Adds a limitation to decompose SDPA until GPU plugin fully supports boolean type attention mask

### Tickets:
 - [CVS-171516](https://jira.devtools.intel.com/browse/CVS-171516)

### CODE REVIEW

File: src/common/transformations/src/transformations/op_conversions/scaled_dot_product_attention_decomposition.cpp [hunk 1]
```diff
@@ -161,7 +161,7 @@ std::shared_ptr<ov::Node> ov::pass::ScaledDotProductAttentionDecomposition::deco
             if (mask.get_element_type() == element::boolean) {
                 atten_mask = register_new_node<v1::ConvertLike>(mask, scaled_atten);
```
Comment (e-ddykim) [inline src/common/transformations/src/transformations/op_conversions/scaled_dot_product_attention_decomposition.cpp]: I think we can remove L162 and L311.

**Review by e-ddykim — APPROVED** [review]

File: src/common/transformations/src/transformations/op_conversions/scaled_dot_product_attention_decomposition.cpp (hunk 1 above)
Comment (andrew-k-park) [reply src/common/transformations/src/transformations/op_conversions/scaled_dot_product_attention_decomposition.cpp]: done

File: src/common/transformations/src/transformations/op_conversions/scaled_dot_product_attention_decomposition.cpp [hunk 2]
```diff
@@ -159,9 +159,8 @@ std::shared_ptr<ov::Node> ov::pass::ScaledDotProductAttentionDecomposition::deco
             // take part in attention. A float mask of the same type as query, key, value that is added to the attention
             // score.
             if (mask.get_element_type() == element::boolean) {
-                atten_mask = register_new_node<v1::ConvertLike>(mask, scaled_atten);
                 auto inv_mask = register_new_node<v1::LogicalNot>(mask);
-                atten_mask = register_new_node<v1::Select>(inv_mask, atten_mask, minus_inf);
+                atten_mask = register_new_node<v1::Select>(inv_mask, minus_inf, zero_f);
```
Comment (mitruska) [inline src/common/transformations/src/transformations/op_conversions/scaled_dot_product_attention_decomposition.cpp]: I would suggest to remove the `LogicalNot` from the line above (L162), use original mask and adjust the "then", "else" order in the following `Select`.
The negation looks redundant here.
```suggestion
                atten_mask = register_new_node<v1::Select>(mask, zero_f, minus_inf);
```

Comment (andrew-k-park) [reply src/common/transformations/src/transformations/op_conversions/scaled_dot_product_attention_decomposition.cpp]: applied

File: src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp [hunk 3]
```diff
@@ -555,6 +555,11 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
             if (sdpa->get_output_element_type(0) != ov::element::f16)
                 return false;
 
+            // - The attn mask type of SDPA should be fp16
+            if (!sdpa->get_causal() && sdpa->get_input_size() >= 4 && sdpa->get_input_element_type(3) != ov::element::f16) {
+                return false;
+            }
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp:561]: nit: what about `element_type == ov::element::boolean` to be more explicit?

Comment (andrew-k-park) [reply src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp:561]: applied

**Review by itikhono — APPROVED** [review]
